### PR TITLE
controller, dto, mappers, etc

### PR DESCRIPTION
**Pull request by OleksiiZaloznyi**

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "books")
+public class BookController {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/bookstore/controller/BookController.java]:
```suggestion
```

**Review by kozhukhovsky — APPROVED** [review]

**Review by Ivan95kos — APPROVED** [review]
Great job

File: src/main/java/com/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -5,32 +5,33 @@
...
-import lombok.RequiredArgsConstructor;
 import org.hibernate.annotations.SQLDelete;
 import org.hibernate.annotations.Where;
-import org.springframework.lang.NonNull;
 
 @Entity
 @Data
 @NoArgsConstructor
-@RequiredArgsConstructor
+//@RequiredArgsConstructor
```
Comment (Ivan95kos) [inline src/main/java/com/example/bookstore/model/Book.java]: uncomment this line

File: src/main/java/com/example/bookstore/model/Book.java [hunk 3]
```diff
@@ -5,32 +5,33 @@
...
 @SQLDelete(sql = "UPDATE book SET is_deleted = true WHERE id=?")
 @Where(clause = "is_deleted=false")
 public class Book {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    @NonNull
+    @NotNull
```
Comment (Ivan95kos) [inline src/main/java/com/example/bookstore/model/Book.java]: It is better to use the `@Column(nullable = false)` for Entity

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Good job overall!

**Review by okuzan — APPROVED** [review]

**Pull request merged**